src/consumers.rs

```diff
@@ -1,31 +1,90 @@
 use std::collections::VecDeque;
 use simulators::Packet;
 
-// Consumer has a VecDeque and its buffer size
-struct Consumer {
+// Consumer stores packets in a queue and processes them
+pub struct Consumer {
     queue: VecDeque<Packet>,
-    buffer_size: usize,
+    buffer_size: Option<usize>,
+    service_time: u32,
+    state: State,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum State {
+    Idle,
+    // Processing holds the time that the consumer will be in processing till
+    Processing(u32),
 }
 
 impl Consumer {
-    fn new(buffer_size: usize) -> Consumer {
+    fn new(buffer_size: Option<usize>, service_time: u32) -> Consumer {
         Consumer {
             queue: VecDeque::new(),
             buffer_size: buffer_size,
+            service_time: service_time,
+            state: State::Idle,
         }
     }
 
-    // If the consumer doesn't have a buffer size (if it's 0), it enques
-    // IF the consumer does have a buffer size, it enques when it's length is less then it's buffer size
-    fn enqueue(&mut self, packet: Packet) -> () {
-        if self.buffer_size == 0 {
-            self.queue.push_back(packet);
-        } else if self.queue.len() < self.buffer_size { 
-            self.queue.push_back(packet);
+    fn enqueue(&mut self, packet: Packet) -> bool {
+        match self.buffer_size {
```

Comment:
Rename `buffer_size` to `buffer_limit` so it reads `buffer_limit` could be `None` or `Some(limit)`.